packages/shared/src/authorization.ts

```diff
@@ -73,19 +80,62 @@ const checkOrgAuthorization: CheckOrgAuthorization = (params, options) => {
   if (!params.role && !params.permission) {
     return null;
   }
+
   if (!orgId || !orgRole || !orgPermissions) {
     return null;
   }
 
   if (params.permission) {
     return orgPermissions.includes(params.permission);
   }
+
   if (params.role) {
     return orgRole === params.role;
   }
   return null;
 };
 
+const checkForFeatureOrPlan = (claim: string, featureOrPlan: string) => {
+  const { org: orgFeatures, user: userFeatures } = splitByScope(claim);
+  const [scope, _id] = featureOrPlan.split(':');
+  const id = _id || scope;
+
+  if (scope === 'org') {
+    return orgFeatures.includes(id);
+  } else if (scope === 'user') {
+    return userFeatures.includes(id);
+  } else {
+    // Since org scoped features will not exist if there is not an active org, merging is safe.
+    return [...orgFeatures, ...userFeatures].includes(id);
+  }
+};
+
+const checkBillingAuthorization: CheckBillingAuthorization = (params, options) => {
+  const { features, plans } = options;
+  if (!params.feature && !params.plan) {
+    return null;
+  }
```

Comment:
[nit] This is redundant. As you are already checking for `params.feature` and `params.plan`  in separate if statements with a fall-through to `null`